src/BlobAdvanced.java

```diff
@@ -88,273 +99,259 @@ void runSamples() throws InvalidKeyException, URISyntaxException, IOException {
 
     /**
      * List containers sample.
-     * @param blobClient Azure Storage Blob Service
+     *
+     * @param blobServiceClient Azure Storage Blob Service
      */
-    private void listContainers(CloudBlobClient blobClient) throws URISyntaxException, StorageException {
+    private void listContainers(BlobServiceClient blobServiceClient) throws URISyntaxException, BlobStorageException {
         ArrayList<String> containerList = new ArrayList<>();
         try {
             System.out.println("Create containers");
             String prefix = UUID.randomUUID().toString();
             for (int i = 0; i < 5; i++) {
                 containerList.add(prefix + i);
-                blobClient.getContainerReference(prefix + i).create();
+                blobServiceClient.createBlobContainer(prefix + i);
             }
 
             System.out.println("List containers");
-            for (final CloudBlobContainer container : blobClient.listContainers(prefix)) {
+            for (final BlobContainerItem container : blobServiceClient.listBlobContainers().stream().filter(blobContainerItem -> blobContainerItem.getName().startsWith(prefix)).collect(Collectors.toList())) {
```

Comment:
You should be able to do blobServiceClient.listBlobContainers(new ListBlobContainersOptions().setPrefix(prefix)) here instead of filtering

Reply:
The code has been fixed according to your suggestion